packages/utils/src/facility/bulk-import-facility.ts

```diff
@@ -0,0 +1,271 @@
+ * - facility-creates.json: is created under runs/import-facility/timestamp it contains the list of facilities that were sent (would have been sent if dryrun) for creation.
+ *
+ * Format of the .csv file:
+ * - first line contains column names
+ * - minimum columns: npi,facilityName,facilityType,cqOboOid,cwOboOid
+ *
+ * Either set the env vars below on the OS or create a .env file in the root folder of this package.
+ *
+ * Execute this with:
+ * $ ts-node src/facility/bulk-import-facility --cx-id <cxId> --timestamp <timestamp> --name <name> --dryrun
+ * $ ts-node src/facility/bulk-import-facility --cx-id <cxId> --timestamp <timestamp> --name <name>
+ */
+
+interface FacilityImportParams {
+  cxId: string;
+  name: string;
+  timestamp: string;
+  dryrun?: boolean;
+}
+
+export const InputRowSchema = z.object({
+  npi: z.string(),
+  facilityName: z.string(),
+  facilityType: z.enum(["obo", "non-obo"]),
+  cqOboOid: z.string().optional(),
+  cwOboOid: z.string().optional(),
+});
+export type InputRowFacilityImport = z.infer<typeof InputRowSchema>;
+
+const S3Utils = getS3UtilsInstance();
+
+const CSV_HEADER =
+  ["npi", "facilityName", "facilityType", "cqOboOid", "cwOboOid", "success", "reason"].join(",") +
+  "\n";
+
+async function main({ cxId, name, timestamp, dryrun }: FacilityImportParams) {
+  await sleep(50); // Give some time to avoid mixing logs w/ Node's
+
+  const bucket = getEnvVarOrFail("FACILITY_IMPORT_BUCKET");
+  const internalUrl = getEnvVarOrFail("API_URL");
+  const isDryRun = Boolean(dryrun);
+
+  console.log(
+    `############## STARTING AT: ${new Date().toISOString()}  ${
+      isDryRun ? "[DRY RUN]" : ""
+    } ##############`
+  );
+
+  const parser = csvParser({
+    headers: ["npi", "facilityName", "facilityType", "cqOboOid", "cwOboOid"],
+    skipLines: 1,
+  });
+
+  const logsFolder = `runs/import-facility/${timestamp}`;
+  const resultFileName = `${name}_result${isDryRun ? "_dryrun" : ""}.csv`;
+  const keyFolder = `ops/facility-import/${cxId}/${timestamp}`;
+
+  const key = `${keyFolder}/${name}`;
+  const logsFilePath = `${logsFolder}/${resultFileName}`;
+  const resultKey = `${keyFolder}/${resultFileName}`;
+  const payloadCreatesFilePath = `${logsFolder}/${name}_facility-creates${
+    isDryRun ? "_dryrun" : ""
+  }.json`;
+
+  await createCsv(logsFilePath, CSV_HEADER);
+
+  const createdFacilities: FacilityInternalDetails[] = [];
+
+  let success = true;
+  let errorMessage: string | undefined;
+
+  const rowPromises: Promise<void>[] = [];
+
+  parser.on("data", async (row: InputRowFacilityImport) => {
+    parser.pause();
+    const p = (async () => {
+      try {
+        const npiFacility = await getFacilityByNpiOrFail(row.npi);
+
+        const ourFacility = translateNpiFacilityToFacilityDetails(npiFacility, row);
+
+        if (!isDryRun) {
+          await createFacility(ourFacility, internalUrl, cxId);
```

Comment:
No need to pass internalUrl as it should be passed directly since the variable will be global